packages/core/src/extensions/SideMenu/SideMenuPlugin.ts

```diff
@@ -283,47 +283,56 @@ export class SideMenuView<
 
     this.editor._tiptapEditor.commands.blur();
 
-    // When ProseMirror handles a drop event on the editor while
-    // `view.dragging` is set, it deletes the selected content. However, if
-    // a block from a different editor is being dropped, this causes some
-    // issues that the code below fixes:
-    if (!this.isDragOrigin && this.pmView.dom.contains(event.target as Node)) {
-      // 1. Because the editor selection is unrelated to the dragged content,
-      // we don't want PM to delete its content. Therefore, we collapse the
-      // selection.
-      this.pmView.dispatch(
-        this.pmView.state.tr.setSelection(
-          TextSelection.create(
-            this.pmView.state.tr.doc,
-            this.pmView.state.tr.selection.to
-          )
-        )
-      );
-    } else if (
-      this.isDragOrigin &&
-      !this.pmView.dom.contains(event.target as Node)
+    let parentEditorElement =
```

Comment:
can we use `closest` here, or not?

If not, on line 289, I think `parentEditorElement !== null` is clearer than `parentEditorElement instanceof HTMLElement`. Reasoning; checking for `instanceof` doesn't match nicely with the if statement of 296. i.e.: if parentElement would return for example a string (which ofc. doesn't happen, but well..), the if statement would pass.

Zoomed out; I don't think we have a cleaner way of checking whether an element is related to a BlockNote editor? Maybe @nperez0111 has thoughts on this

Reply:
I'm unsure of this, but the way that this was solved in Tiptap was to store the currently dragged editor instance in a global var, so it could be distinguished whether the drag came from another instance.

https://github.com/ueberdosis/tiptap/pull/5893/files